libs/remix-ui/clipboard/src/lib/copy-to-clipboard/copy-to-clipboard.tsx

```diff
@@ -6,30 +6,41 @@ import { Placement } from 'react-bootstrap/esm/Overlay'
 import './copy-to-clipboard.css'
 
 interface ICopyToClipboard {
-  content: any,
+  content?: any,
   tip?: string,
   icon?: string,
   direction?: Placement,
   className?: string,
   title?: string,
-  children?: JSX.Element
+  children?: JSX.Element,
+  getContent?: () => {}
 }
 export const CopyToClipboard = (props: ICopyToClipboard) => {
-  let { content, tip = 'Copy', icon = 'fa-copy', direction = 'right', children, ...otherProps } = props
+  let { content, tip = 'Copy', icon = 'fa-copy', direction = 'right', children, getContent, ...otherProps } = props
   const [message, setMessage] = useState(tip)
-  const handleClick = (e) => {
-    if (content && content !== '') { // module `copy` keeps last copied thing in the memory, so don't show tooltip if nothing is copied, because nothing was added to memory
-      try {
-        if (typeof content !== 'string') {
-          content = JSON.stringify(content, null, '\t')
-        }
-        copy(content)
-        setMessage('Copied')
-      } catch (e) {
-        console.error(e)
+
+  const copyData = () => {
+    try {
+      if (content === '') {
+        setMessage('Cannot copy empty content!')
+        return
+      }
+      if (typeof content !== 'string') {
+        content = JSON.stringify(content, null, '\t')
       }
+      copy(content)
+      setMessage('Copied')
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  const handleClick = (e) => {
+    if (content) { // module `copy` keeps last copied thing in the memory, so don't show tooltip if nothing is copied, because nothing was added to memory
+      copyData()
     } else {
-      setMessage('Cannot copy empty content!')
+      content = getContent && getContent()
+      copyData()
     }
     e.preventDefault()
     return false
```

Comment:
Can you remove think line? I don't think it is needed anymore.

Reply:
done.